test/integration/envtest/shootmaintenance/maintenance_control_test.go

```diff
@@ -0,0 +1,321 @@
+							},
+							{
+								ExpirableVersion: gardencorev1beta1.ExpirableVersion{
+									Version:        testMachineImageVersion,
+									Classification: &deprecatedClassification,
+								},
+							},
+						},
+					},
+				},
+				MachineTypes: []gardencorev1beta1.MachineType{
+					{
+						Name: "large",
+					},
+				},
+				Regions: []gardencorev1beta1.Region{
+					{
+						Name: "foo-region",
+					},
+				},
+				Type: "foo-type",
+			},
+		}
+		Expect(testClient.Create(ctx, cloudProfile)).To(Succeed())
+
+		By("create shoot")
+		shoot = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-shoot", Namespace: "garden-dev"},
+			Spec: gardencorev1beta1.ShootSpec{
+				SecretBindingName: "my-provider-account",
+				CloudProfileName:  "test-cloudprofile",
+				Region:            "foo-region",
+				Provider: gardencorev1beta1.Provider{
+					Type: "foo-provider",
+					Workers: []gardencorev1beta1.Worker{
+						{
+							Name:    "cpu-worker",
+							Minimum: 2,
+							Maximum: 2,
+							Machine: gardencorev1beta1.Machine{
+								Image: &gardencorev1beta1.ShootMachineImage{
+									Name:    "foo-image",
+									Version: pointer.String("1.1.1"),
+								},
+								Type: "large",
+							},
+						},
+					},
+				},
+				Kubernetes: gardencorev1beta1.Kubernetes{
+					Version: "1.21.1",
+				},
+				Networking: gardencorev1beta1.Networking{
+					Type: "foo-networking",
+				},
+				Maintenance: &gardencorev1beta1.Maintenance{
+					AutoUpdate: &gardencorev1beta1.MaintenanceAutoUpdate{
+						KubernetesVersion:   false,
+						MachineImageVersion: false,
+					},
+				},
+			},
+		}
+
+		// remember highest version of the image.
+		highestShootMachineImage = *shoot.Spec.Provider.Workers[0].Machine.Image
+		// set dummy kubernetes version to shoot
+		shoot.Spec.Kubernetes.Version = testKubernetesVersionLowPatchLowMinor.Version
+		// set shoot MachineImage version to testMachineImage
+		shoot.Spec.Provider.Workers[0].Machine.Image.Version = testMachineImage.Version
+		// remember the test machine image
+		// also required to know for which image name the test versions should be added to the CloudProfile
+		testMachineImage = *shoot.Spec.Provider.Workers[0].Machine.Image
+		Expect(testClient.Create(ctx, shoot)).To(Succeed())
+	})
+
+	AfterEach(func() {
+		logger.Infof("Delete shoot %s", shoot.Name)
+		Expect(DeleteShoot(ctx, testClient, shoot)).To(Succeed())
+
+		logger.Infof("Delete CloudProfile %s", cloudProfile.Name)
+		Expect(testClient.Delete(ctx, cloudProfile)).To(Succeed())
+	})
+
+	Describe("Machine image maintenance tests", func() {
+		It("Do not update Shoot machine image in maintenance time: AutoUpdate.MachineImageVersion == false && expirationDate does not apply", func() {
+			err := kutil.SetAnnotationAndUpdate(ctx, testClient, shoot, v1beta1constants.GardenerOperation, v1beta1constants.ShootOperationMaintain)
+			Expect(err).To(BeNil())
```

Comment:
```suggestion
			Expect(kutil.SetAnnotationAndUpdate(ctx, testClient, shoot, v1beta1constants.GardenerOperation, v1beta1constants.ShootOperationMaintain)).To(Succeed())
```

You could apply the same for other occurrences.